android/src/main/java/com/oney/WebRTCModule/PeerConnectionObserver.java

```diff
@@ -528,4 +550,94 @@ private String signalingStateString(PeerConnection.SignalingState signalingState
         }
         return null;
     }
+
+    /**
+     * Serialization and Parsing helpers
+     */
+
+    private WritableMap serializeStream(String streamReactTag, MediaStream stream) {
+
+        WritableMap params = Arguments.createMap();
+        params.putInt("id", id);
+        params.putString("streamId", stream.getId());
+        params.putString("streamReactTag", streamReactTag);
+
+        WritableArray tracks = Arguments.createArray();
+
+        for (VideoTrack track: stream.videoTracks) {
+            tracks.pushMap(serializeTrack(track));
+        }
+        for (AudioTrack track: stream.audioTracks) {
+            tracks.pushMap(serializeTrack(track));
+        }
+
+        params.putArray("tracks", tracks);
+
+        return params;
+    }
+
+    public String serializeDirection(RtpTransceiver.RtpTransceiverDirection src) {
+        switch(src) {
+            case INACTIVE:
+                return "inactive";
+            case RECV_ONLY:
+                return "recvonly";
+            case SEND_ONLY:
+                return "sendonly";
+            case SEND_RECV:
+                return "sendrecv";
+            default:
+                throw new Error("Invalid direction");
+        }
+    }
+
+    private ReadableMap serializeTrack(MediaStreamTrack track) {
+        WritableMap trackInfo = Arguments.createMap();
+        trackInfo.putString("id", track.id());
+        trackInfo.putInt("peerConnectionId", this.id);
+        trackInfo.putString("kind", track.kind());
+        trackInfo.putBoolean("enabled", track.enabled());
+        trackInfo.putString("readyState", track.state().toString().toLowerCase());
+        trackInfo.putBoolean("remote", true);
+        return trackInfo;
+    }
+
+    /**
+     * This method is currently missing serializing DtmfSender
+     * and transport.
+     * TODO: Add transport and dtmf fields to the serialized sender to match the web APIs.
+     */
+    public ReadableMap serializeSender(RtpSender sender) {
+        WritableMap res = Arguments.createMap();
+        res.putString("id", sender.id());
+        res.putInt("peerConnectionId", this.id);
+        if (sender.track() != null)
+            res.putMap("track", serializeTrack(sender.track()));
+        return res;
+    }
+
+    private ReadableMap serializeReceiver(RtpReceiver receiver) {
+        WritableMap res = Arguments.createMap();
+        res.putString("id", receiver.id());
+        res.putInt("peerConnectionId", this.id);
+        res.putMap("track", serializeTrack(receiver.track()));
+        return res;
+    }
+
+    public ReadableMap serializeTransceiver(RtpTransceiver transceiver) {
```

Comment:
That would be better, but the reason why it is inside the observer is because it has access to the peerConnectionId, it does not matter that much as we can provide it as an argument to the functions. What do you reckon?

Reply:
A parameter sounds good!

Reply:
Done.